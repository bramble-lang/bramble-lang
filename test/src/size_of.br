fn my_main() -> i64 {
    project::std::io::writeu64ln(size_of(i64));
    project::std::io::writeu64ln(size_of(i32));
    project::std::io::writeu64ln(size_of(i16));
    project::std::io::writeu64ln(size_of(i8));
    project::std::io::writeu64ln(size_of(u64));
    project::std::io::writeu64ln(size_of(u32));
    project::std::io::writeu64ln(size_of(u16));
    project::std::io::writeu64ln(size_of(u8));
    project::std::io::writeu64ln(size_of(bool));
    project::std::io::writeu64ln(size_of(f64));
    project::std::io::writeu64ln(size_of([f64; 3]));
    project::std::io::writeu64ln(size_of(MyStruct));
    project::std::io::writeu64ln(size_of(MyStruct2));
    project::std::io::writeu64ln(size_of(*const MyStruct2));
    project::std::io::writeu64ln(size_of(*mut MyStruct2));

    return 0;
}

struct MyStruct {
    b: bool,
    c: i8,
}

struct MyStruct2 {
    x: i64,
    y: f64,
}