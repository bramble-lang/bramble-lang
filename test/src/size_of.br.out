8
4
2
1
8
4
2
1
1
8
24
2
16
